src/mailer/index.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * Mailer module
+ *
+ * This is the entrypoint of the mailer module, which is spread over multiple files in this directory.
+ * Ideally this is the only file that's needed to be imported when apps want to send email.
+ */
+
+import { fromNullable, none, Option, some } from "fp-ts/lib/Option";
+import { agent } from "italia-ts-commons";
+import {
+  AbortableFetch,
+  setFetchTimeout,
+  toFetch
+} from "italia-ts-commons/lib/fetch";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import { Millisecond } from "italia-ts-commons/lib/units";
+
+import {
+  MailerConfig,
+  MailhogMailerConfig,
+  MailupMailerConfig,
+  MultiTrasnsportMailerConfig,
+  SendgridMailerConfig
+} from "./config";
+
+import {
+  createMailTransporter,
+  getTransportsForConnections,
+  MailerTransporter,
+  MailUpTransport,
+  MultiTransport,
+  NodeMailerSendgrid,
+  Transport
+} from "./transports";
+
+// expects a never value. return a constant or the value itself
+const defaultNever = <T>(e: never, retVal: T = e): T => retVal;
+
+// 5 seconds timeout by default
+const DEFAULT_EMAIL_REQUEST_TIMEOUT_MS = 5000;
+
+// Must be an https endpoint so we use an https agent
+const abortableFetch = AbortableFetch(agent.getHttpsFetch(process.env));
+const fetchWithTimeout = setFetchTimeout(
+  DEFAULT_EMAIL_REQUEST_TIMEOUT_MS as Millisecond,
+  abortableFetch
+);
+const fetchAgent = toFetch(fetchWithTimeout);
```

Comment:
I think, since this is accessing process.env, this should be configured in the caller and removed from this module but passed as a paramtere to `getMailerTransporter`